exporter/opentelemetry-exporter-otlp/src/opentelemetry/exporter/otlp/exporter.py

```diff
@@ -31,11 +34,17 @@
 
 from opentelemetry.proto.common.v1.common_pb2 import AnyValue, KeyValue
 from opentelemetry.proto.resource.v1.resource_pb2 import Resource
+from opentelemetry.sdk.resources import Resource as SDKResource
 
 logger = logging.getLogger(__name__)
+SDKDataT = TypeVar("SDKDataT")
+ResourceDataT = TypeVar("ResourceDataT")
+TypingResourceT = TypeVar("TypingResourceT")
```

Comment:
I believe these are the only two possibilities at the moment, so you can constrain this:

https://github.com/open-telemetry/opentelemetry-python/blob/1abdc02a92cd996f3924e1653896bb72123ac84e/opentelemetry-proto/src/opentelemetry/proto/metrics/v1/metrics_pb2.pyi#L55

https://github.com/open-telemetry/opentelemetry-python/blob/1abdc02a92cd996f3924e1653896bb72123ac84e/opentelemetry-proto/src/opentelemetry/proto/trace/v1/trace_pb2.pyi#L54

Reply:
Related to above comment on constraining the type here.